crates/test-programs/src/bin/tls_sample_application.rs

```diff
@@ -28,19 +27,32 @@ fn test_tls_sample_application() {
         .unwrap();
 
     let (client_connection, tls_input, tls_output) =
-        ClientHandshake::new(DOMAIN, tcp_input, tcp_output)
+        ClientHandshake::new(domain, tcp_input, tcp_output)
             .blocking_finish()
             .unwrap();
 
     tls_output.blocking_write_util(request.as_bytes()).unwrap();
     client_connection
         .blocking_close_output(&tls_output)
         .unwrap();
-    socket.shutdown(ShutdownType::Send).unwrap();
+    socket.shutdown(ShutdownType::Send)?;
     let response = tls_input.blocking_read_to_end().unwrap();
-    let response = String::from_utf8(response).unwrap();
+    String::from_utf8(response).context("error converting response")
+}
 
-    assert!(response.contains("HTTP/1.1 200 OK"));
+fn test_tls_sample_application() {
+    // since this is testing remote endpoint to ensure system cert store works
+    // the test uses a couple different endpoints to reduce the number of flakes
+    const DOMAIN1: &'static str = "example.com";
+    const DOMAIN2: &'static str = "api.github.com";
+
+    let response1 = make_tls_request(DOMAIN1).unwrap();
+    let response2 = make_tls_request(DOMAIN2).unwrap();
+
```

Comment:
Should this avoid a `.unwrap()` here since I think that would translate to both domains being required to succeed instead of just one?

Reply:
yes, I had that at initially and was trying something else and forgot to go back🤦

Reply:
updated